### PR TITLE
Implement AERI preferences & wizards

### PR DESCRIPTION
**Pull request by rawagner**

### CODE REVIEW

**Comment by rawagner** [comment]
testPR

File: plugins/org.jboss.reddeer.jface/src/org/jboss/reddeer/jface/matcher/DialogWithTitleMatcher.java [hunk 1]
```diff
@@ -10,61 +10,48 @@
...
-				Control labelControl = compositeChildren[2];
-				if(labelControl instanceof Label){
-					String labelText = lHandler.getText((Label)labelControl);
-					return matcher.matches(labelText);
-				}
-			}catch (Exception e) {
-				return false;
+			if(new WindowMatcher(org.eclipse.jface.dialogs.TitleAreaDialog.class).matches(item)){
+				DefaultShell shell = new DefaultShell((Shell)item);
+				TitleAreaDialog ta = new TitleAreaDialog(shell);
```
Comment (mlabuda) [inline plugins/org.jboss.reddeer.jface/src/org/jboss/reddeer/jface/matcher/DialogWithTitleMatcher.java]: what if something goes wrong here with SWT widget, e.g. gets disposed? It fails on exception - is this desired?

**Comment by rawagner** [comment]
@mlabuda fixed also for WindowMatcher

File: plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/epp/logging/aeri/ide/dialogs/PreferencePage.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+import org.eclipse.reddeer.swt.impl.combo.LabeledCombo;
+import org.eclipse.reddeer.swt.impl.group.DefaultGroup;
+import org.eclipse.reddeer.swt.impl.text.LabeledText;
+
+/**
+ * Represends AERI preference page
+ * @author rawagner
+ *
+ */
+public class PreferencePage extends org.eclipse.reddeer.jface.preference.PreferencePage{
```
Comment (apodhrad) [inline plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/epp/logging/aeri/ide/dialogs/PreferencePage.java]: I suggest to use other name, for example AeriPreferencePage

File: plugins/org.eclipse.reddeer.swt/src/org/eclipse/reddeer/swt/api/Label.java [hunk 3]
```diff
@@ -10,8 +10,8 @@
  *******************************************************************************/
 package org.eclipse.reddeer.swt.api;
 
-
 import org.eclipse.swt.graphics.Image;
+
```
Comment (apodhrad) [inline plugins/org.eclipse.reddeer.swt/src/org/eclipse/reddeer/swt/api/Label.java]: Wow... what a change :)

File: plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/epp/logging/aeri/ide/dialogs/PreferencePage.java (hunk 2 above)
Comment (rawagner) [inline plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/epp/logging/aeri/ide/dialogs/PreferencePage.java]: That is how aeri preference page is called in Eclipse

File: plugins/org.eclipse.reddeer.swt/src/org/eclipse/reddeer/swt/api/Label.java (hunk 3 above)
Comment (rawagner) [inline plugins/org.eclipse.reddeer.swt/src/org/eclipse/reddeer/swt/api/Label.java]: well...not it looks much better then before :)

Comment (apodhrad) [inline plugins/org.eclipse.reddeer.swt/src/org/eclipse/reddeer/swt/api/Label.java]: That's why I asked you for code formatting in other classes (we will avoid such changes in the future).

Comment (rawagner) [inline plugins/org.eclipse.reddeer.swt/src/org/eclipse/reddeer/swt/api/Label.java]: formatted all classes that I changed or created

**Comment by apodhrad** [comment]
**Failed**
```
org.eclipse.reddeer.eclipse.test.epp.logging.aeri.ide.dialogs.AeriTest.testAeriServers default
Failing for the past 1 build (Since Unstable#388 )
Took 2.9 sec.
Error Message

Unable to execute action because modal shell '' exists
```
**Stacktrace**
```
org.eclipse.reddeer.core.exception.CoreLayerException: Unable to execute action because modal shell '' exists
	at org.eclipse.reddeer.core.handler.ControlHandler.checkModalShells(ControlHandler.java:172)
	at org.eclipse.reddeer.core.handler.ShellHandler.setFocus(ShellHandler.java:75)
	at org.eclipse.reddeer.swt.impl.shell.AbstractShell.setFocus(AbstractShell.java:63)
	at org.eclipse.reddeer.swt.widgets.AbstractControl.<init>(AbstractControl.java:55)
	at org.eclipse.reddeer.swt.impl.shell.AbstractShell.<init>(AbstractShell.java:35)
	at org.eclipse.reddeer.swt.impl.shell.DefaultShell.<init>(DefaultShell.java:38)
	at org.eclipse.reddeer.jface.matcher.DialogWithTitleMatcher.matches(DialogWithTitleMatcher.java:53)
	at org.eclipse.reddeer.common.matcher.AndMatcher.matches(AndMatcher.java:49)
	at org.eclipse.reddeer.core.condition.ShellMatchingMatcherIsAvailable.test(ShellMatchingMatcherIsAvailable.java:48)
	at org.eclipse.reddeer.common.wait.WaitUntil.stopWaiting(WaitUntil.java:89)
	at org.eclipse.reddeer.common.wait.AbstractWait.wait(AbstractWait.java:131)
	at org.eclipse.reddeer.common.wait.AbstractWait.<init>(AbstractWait.java:116)
	at org.eclipse.reddeer.common.wait.AbstractWait.<init>(AbstractWait.java:81)
	at org.eclipse.reddeer.common.wait.AbstractWait.<init>(AbstractWait.java:63)
	at org.eclipse.reddeer.common.wait.WaitUntil.<init>(WaitUntil.java:49)
	at org.eclipse.reddeer.core.lookup.ShellLookup.getShell(ShellLookup.java:155)
	at org.eclipse.reddeer.core.lookup.ShellLookup.getShell(ShellLookup.java:171)
	at org.eclipse.reddeer.jface.window.AbstractWindow.<init>(AbstractWindow.java:70)
	at org.eclipse.reddeer.jface.dialogs.TitleAreaDialog.<init>(TitleAreaDialog.java:51)
	at org.eclipse.reddeer.eclipse.epp.logging.aeri.ide.dialogs.ConfigureServerDialog.<init>(ConfigureServerDialog.java:34)
	at org.eclipse.reddeer.eclipse.epp.logging.aeri.ide.dialogs.ReportingProject.configure(ReportingProject.java:47)
	at org.eclipse.reddeer.eclipse.test.epp.logging.aeri.ide.dialogs.AeriTest.testAeriServers(AeriTest.java:54)
```

**Comment by rawagner** [comment]
rebased on latest master, the failure is from modal shells check which was removed

**Comment by apodhrad** [comment]
Yes, that's the issue (I have verified it locally).

**Pull request merged**